mlflow/models/model.py

```diff
@@ -716,7 +716,11 @@ def log(
                     except Exception as e:
                         _logger.warning("Failed to load model config from %s: %s", model_config, e)
                 try:
-                    mlflow.tracking.fluent.log_params(model_config or {}, run_id=run_id)
+                    from mlflow.models.utils import _flatten_nested_params
+
+                    mlflow.tracking.fluent.log_params(
+                        _flatten_nested_params(model_config) or {}, run_id=run_id
```

Comment:
can we pass the `sep="/"` explicitly here so its more readable how this works? and can we also add a comment here on what we are doing and why?